pkg/cloud/services/networking/network.go

```diff
@@ -58,15 +58,19 @@ func (s *Service) ReconcileNetwork(clusterName string, openStackCluster *infrav1
 	}
 
 	var portSecurityEnabled gophercloud.EnabledState
+	var opts createOpts
 	if openStackCluster.Spec.DisablePortSecurity {
 		portSecurityEnabled = gophercloud.Disabled
+		opts = createOpts{
+			AdminStateUp:        gophercloud.Enabled,
+			Name:                networkName,
+			PortSecurityEnabled: portSecurityEnabled,
+		}
 	} else {
-		portSecurityEnabled = gophercloud.Enabled
-	}
-	opts := createOpts{
-		AdminStateUp:        gophercloud.Enabled,
-		Name:                networkName,
-		PortSecurityEnabled: portSecurityEnabled,
+		opts = createOpts{
+			AdminStateUp: gophercloud.Enabled,
+			Name:         networkName,
```

Comment:
if I understand correctly, the logic is if we should disable port security
then we should not provide this param to gophercloud and in turn not to opentsack neutron?

so we need 2 flag one is to indicate whether the port security will be used and the other is for whether the network itself need port security or not 

so better to rename the variable to `SecurityGroupEnabled`? so that we only have 2 options
if enbale security group, then mandatory to make the network enable security group and if not enable security group extension, then not put param here?

Reply:
The other way around, if we want to disable port security we pass that parameter to gophercloud and neutron. If we don't want to disable port security we don't pass the parameter.As far as I can tell from the documentation I can find networks are by default created with port security enabled.

https://wiki.openstack.org/wiki/Neutron/ML2PortSecurityExtensionDriver

Reply:
Not sure what's going on without the extension. But when the extension is enabled the default is true: https://docs.openstack.org/api-ref/network/v2/ (search for `port_security_enabled`)

So imho the PR is fine. If we want to disable it disable it, if not, just don't set the attribute and the default value is used and port security is enabled

Reply:
@qvicksilver one nit otherwise /approve from me

Reply:
@sbueringer Looks sane to me!